### ISSUE TITLE
Fix S2201 FN: Support native ints

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#6399]
### Description

FN for S2201 (ReturnValueIgnored).

Using C#11 native ints, the rule is not raising an issue.

### Repro steps

```csharp
 using System;

int x = 2;

nint n = 2;
nuint nu = 2;

IntPtr intPtr = 2;
UIntPtr uintPtr = 2;

n.ToInt32(); // Compliant, FN, pure function
nu.ToUInt32(); // Compliant, FN, pure function

intPtr.ToInt64(); // Compliant, FN, pure function
uintPtr.ToUInt64(); // Compliant, FN, pure function

unsafe
{
    n.ToPointer(); // Compliant, FN, pure function
    intPtr.ToPointer(); // Compliant, FN, pure function


    nu.ToPointer(); // Compliant, FN, pure function
    uintPtr.ToPointer(); // Compliant, FN, pure function

    x.ToString(); // Noncompliant
}
```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule does not raise an issue.
